spring-data-relational/src/main/java/org/springframework/data/relational/core/dialect/OrderByOptionsSupport.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.relational.core.dialect;
+
+import org.springframework.data.domain.Sort;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+
+/**
+ * The {@code ORDER BY} options supported by a specific Dialect.
+ *
+ * @author Chirag Tailor
+ */
+public interface OrderByOptionsSupport {
```

Comment:
I'd like to have a clearer comment on this. And on the method as well.

As it currently stands I would have expected a list of enum values listing all the supported options.
The comments should also explain what a "Order By Option" is.